### PR TITLE
Moving packaged_task to module futures

### PR DESCRIPTION
**Pull request by hkaiser**
- flyby: move tests related to local futures to futures module

### CODE REVIEW

File: libs/core/futures/tests/unit/local_use_allocator.cpp [hunk 1]
```diff
@@ -32,4 +32,6 @@ int main()
                       test_allocator<int>>::value,
         "std::uses_allocator<local::packaged_task<int()>, "
         "test_allocator<int> >::value");
+
+    return 0;
```
Comment (msimberg) [inline libs/core/futures/tests/unit/local_use_allocator.cpp:36]: Not really necessary.

**Review by msimberg — COMMENTED** [review]
Is the move just a preference or is there a more fundamental reason to put it in the futures module?

File: libs/core/futures/tests/unit/local_use_allocator.cpp (hunk 1 above)
Comment (hkaiser) [reply libs/core/futures/tests/unit/local_use_allocator.cpp:36]: I know :/ I usually add it for consistency anyways.

Comment (msimberg) [reply libs/core/futures/tests/unit/local_use_allocator.cpp:36]: I suspected you knew :P Maybe someone who didn't know before knows now... I don't mind keeping the explicit return.

**Comment by hkaiser** [comment]
This is ready to go in.

**Comment by msimberg** [comment]
I've nothing against this changed (it's merged), but I'm still looking for an answer to this.

> Is the move just a preference or is there a more fundamental reason to put it in the futures module?

**Comment by hkaiser** [comment]
> I've nothing against these changes but I'm still looking for an answer to this:
> 
> > Is the move just a preference or is there a more fundamental reason to put it in the futures module?

Sorry, I didn't notice your question. There are two reasons that motivated the change:
- packaged_task belongs into the same group of types as future and promise, so I thought we should have it moved
- during some experimentation with executors I wanted to use `packaged_task`, but that created circular dependencies with the executor module - moving the file helped resolving those

**Pull request merged**